editor/plugins/script_text_editor.cpp

```diff
@@ -1732,39 +1765,65 @@ void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data
 	int row = pos.y;
 	int col = pos.x;
 
+	const bool use_type = EDITOR_GET("text_editor/completion/add_type_hints");
```

Comment:
This should only be set when `drop_modifier_pressed` is true (because otherwise it's unused). I think it was fine defined where it was before.